src/core/activityContent/validation.test.ts

```diff
@@ -175,17 +293,82 @@ describe("activity content validations", () => {
               },
             ],
           },
-        ],
-      },
-      "a note with an video attachement with a non-array URL field": {
-        "@context": "https://www.w3.org/ns/activitystreams",
-        type: "Note",
-        content: "What an adventure!",
-        mediaType: "text/plain",
-        attachment: [
-          {
-            type: "Video",
-            url: {
+        ];
+        expect(requireGetSupportedContentAttachments(multipleUrlsAttachment)).toStrictEqual(multipleUrlsAttachment);
+      });
+
+      it("with multiple attachments but only one is usable/valid, returns the valid one", () => {
+        const validAttachment = {
+          type: "Image",
+          url: [
+            {
+              type: "Link",
+              href: "https://upload.wikimedia.org/wikipedia/commons/a/ae/Mccourt.jpg",
+              mediaType: "image/jpg",
+              hash: [
+                {
+                  algorithm: "keccak256",
+                  value: "0x90b3b09658ec527d679c2de983b5720f6e12670724f7e227e5c360a3510b4cb5",
+                },
+              ],
+            },
+          ],
+        };
+        const invalidAttachment = {
+          type: "Image",
+          url: [
+            {
+              type: "Link",
+              href: "https://upload.wikimedia.org/wikipedia/commons/a/ae/Mccourt.jpg",
+              hash: [
+                {
+                  algorithm: "keccak256",
+                  value: "0x90b3b09658ec527d679c2de983b5720f6e12670724f7e227e5c360a3510b4cb5",
+                },
+              ],
+            },
+          ],
+        };
+
+        expect(requireGetSupportedContentAttachments([validAttachment, invalidAttachment])).toHaveLength(1);
+      });
+
+      it("with multiple link attachments and one is invalid, returns the valid one", () => {
```

Comment:
I know these seem really repetitive, but the code path is slightly different when it's a link attachment versus an Image, Video, or Audio attachment.  I added the attachment type to the other descriptions.

Reply:
in that case, maybe we can change the descriptions to clarify the difference? something like `it("returns valid attachments for Link type attachments")` here?

Reply:
I realized I should not have put the test cases for a single attachment with multiple URLs in the "multiple attachments."  test suite. This is going to be a source of confusion I think for everyone.  Anyway I  moved them into their proper test suites and adjusted descriptions.

Reply:
And also in that process I discovered a new error case and added a couple of tests for that. So if there is a single attachment which fails the type checks, it throws regardless and the attachment is invalid.  Also I added some clarifying examples in `requireGetSupportedContentAttachments`  as to when one should expect something to throw vs. returning validated attachments.   This seems like it would be the most desirable behavior to me so lmk if you disagree.